### PR TITLE
DR_0004_questionnaire-change-rules

### PR DESCRIPTION
**Pull request by dpalomino**
Hi, 

I've created a draft proposal containing some rules for allowing some minimum changes when updating questionnaires in existing projects with actual data. I've tried to be conservative and only propose to allow simple changes, avoiding any modifications related with type change, field name change or removing things. 

Please, let me know your thoughts.

### CODE REVIEW

File: DR_0004_questionnaire-change-rules.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+The following is an exhaustive list of the allowed changes when updating a questionnaire (i.e. uploading a new version of it) once that there is some data in the project. Anything that is not listed in the following list is not allowed. If there is no data yet in the project, then any change in the questionnaire will be allowed.
+
+List of allowed changes: 
+1. Changes in `form_id` and `title` (in the `settings` tab)
+2. Change in the default language to use (`default_language` in `settings` tab)
+3. Add new languages for the labels (i.e. adding `label:xx` column in `survey` and `choices` tab)
+4. Change in language used in labels
+5. Removal of a language used in labels (i.e. deleting `label:xx` column in `survey`and `choices` tab)
+6. Any change in any label in either `survey` or `choices`
+7. Adding new choices for the multiple-choice questions in `choices` tab, with the exception of `respondent`, `land_type`, `tenure_type` and `geo_type`. These will not accept any changes for now (we could revisit this when adding support for additional `respondent`, `land_type` and `tenure_type).
```
Comment (oliverroick) [inline DR_0004_questionnaire-change-rules.md]: There's a tiny syntax error at the end of this line: "`" is missing.

**Review by oliverroick — COMMENTED** [review]
I think this list is reasonable. 

One question: How do we handle cases when users upload a questionnaire with a change that is not allowed? Do we reject the form completely or do we ignore changes that are not allowed? I'm leaning towards rejecting the form and providing an error message outlining the changes that are allowed.

**Comment by dpalomino** [comment]
Hi @oliverroick! Thanks for fixing the typo and for the feedback :-) Yes, I also think that it could be better to reject the questionnaire providing an error message explaining what is and is not allowed. Doing the opposite could be a bit confusing to the user I think, and perhaps user would like to include other modifications after seeing the changes that have been not allowed.

**Review by dpalomino — COMMENTED** [review]
Format edits

**Comment by amplifi** [comment]
Feedback from dev team meeting (H/T @bjohare): Should specifically note that an optional field cannot be later made required.

**Comment by dpalomino** [comment]
Thanks @bjohare! I've added that case, explicitly saying that we are not allowing an optional field becoming mandatory. The opposite case (mandatory becoming optional) IS allowed, with the exemption of `party_type`, `party_name`, `location_type`, and `tenure_type`.

**Pull request merged**